pkg/diff/diffbycommand.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2024 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package diff
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+
+	"sigs.k8s.io/yaml"
+)
+
+// DiffByCommand converts the given objects into yaml and then runs the command to compare them.
+func DiffByCommand(command string, old, new interface{}) ([]byte, error) {
+	oldBytes, err := yaml.Marshal(old)
```

Comment:
[nit] Check whether the command exists or not.

Reply:
thanks, I fixed it. [c9764e4](https://github.com/pipe-cd/pipecd/pull/4881/commits/c9764e436d27c8dca67789f6b31bcae507a56e0a)